src/apps/search-result/search-result.tsx

```diff
@@ -28,33 +27,34 @@ import FacetLine from "../../components/facet-line/FacetLine";
 import { getUrlQueryParam } from "../../core/utils/helpers/url";
 import useGetCleanBranches from "../../core/utils/branches";
 import { dataIsNotEmpty } from "../../core/utils/helpers/general";
+import { RootState, useSelector } from "../../core/store";
+import { addToFilter } from "../../core/filter.slice";
 
 interface SearchResultProps {
   q: string;
   pageSize: number;
 }
 
 const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
+  const dispatch = useDispatch();
+  const filterState = useSelector((state: RootState) => state.filter);
```

Comment:
This gets the filters directly from the store but there does not seem to be any checks that confirm that the stored filters actually match the query. This results in the following behavior:

1. User searches for "Harry Potter"
2. User filters on facet term "Magic"
3. User goes to material "Harry Potter and the Philosophers Stone"
4. User goes back to the search result for "Harry Potter"
5. The result will be filtered by the facet term "Magic"
6. The user goes back to the material "Harry Potter and the Philosophers Stone"
7. The user follows a link to the search result for "Fiction" e.g. by clicking on a term.
8. The search result will be filtered by the facet term "Magic"

While the filter in 5 is intentional the filter in 8 is not.

We need a way to distinguish the two and clear the stored filters if a new query is made.